scripts/pipeline/appropriate_ref_collection_fasta.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classifies the appropriate set of comparators for a taxon by taxonomic-directory
+
+Example::
+
+    $ python appropriate_ref_collection_fasta.py --taxon_name canis_lupis_familiaris \
+        --genome_fasta ${ENSEMBLFTP}/Canis_lupus_familiaris/Canis_lupus_familiaris.fasta \
+        --url mysql://ensro@mysql-ens-sta-1:4519/ncbi_taxonomy_106 \
+        --taxon_name canis_lupis_familiaris \
+        --comparators_dir /hps/nobackup/flicek/ensembl/compara/shared/references_symlink_fasta \
+        --dest_dir '45.88.81.155:/bucket3'
+
+"""
+
+import os
+from os.path import basename
+from pathlib import Path
+from shutil import copytree, copyfile
+import sys
+
+from argparse import ArgumentParser
+
+from sqlalchemy.orm import Session
+
+from ensembl.database import DBConnection
+from ensembl.compara.utils.taxonomy import (
+    collect_taxonomys_from_path,
+    match_taxon_to_reference,
+)
+
+
+def _get_collection_path(session: Session, taxon: str, ref_dir: Path) -> Path:
+    """Returns an appropriate reference collection directory for query ``taxon``
+
+    Args:
+        taxon: Scientific taxon name as in ncbi_taxonomy
+        dir: collections directory containing directories
+            named by taxonomic classification
+    """
+    refs_dir = collect_taxonomys_from_path(session, ref_dir)
+    ref_taxon = match_taxon_to_reference(session, taxon, refs_dir)
+    return os.path.join(ref_dir, ref_taxon)
+
+
+def _duplicate_collection_path(taxon: str, src: Path, dest: Path) -> Path:
+    """Copies given directory to given location and renames for query ``taxon``
+
+    Args:
+        taxon: Scientific taxon name as in ncbi_taxonomy
+        src: Collections directory named by taxonomic classification
+        dest: Path of new directory parent
+    """
+    dest = os.path.join(dest, taxon)
+    try:
+        new_collection = copytree(src, dest)
+        return new_collection
+    except FileExistsError:
```

Comment:
Is it possible that the dest directory exists but the references are not the same? If this is the case you would process the data with the wrong references set?

Reply:
It should not be possible in our very specific use-case for this script. Although I am not sure what else this script would be used for. It depends on if we would use this script for any other purpose than to be run in a workflow in which the dest dir is a temporary dir as it is in this case (as it heads to short-term storage). To note than in our use-case the dest dir is to be removed near the end of the pipeline - it is an intermediate step.